### PR TITLE
Add new select platform for forced options

### LINKED ISSUE #78: Implement forced option as select entity

### PR DESCRIPTION
**Pull request by kingsleyadam**
Closes #78 

This adds the new platform `select` with the forced positions for SwitchActuator, CoverActuator, and DimmingActuator.

Some work on the python package side will still need to be required before this can be merged in, marking this as a draft for now.

I've kept an option to set `available` property up to the Python package, fyi @derjoerg, if the python package implements `available` then it'll be inherited by Home Assistant entities.

### CODE REVIEW

File: custom_components/abbfreeathome_ci/select.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+
+class FreeAtHomeSelectEntity(SelectEntity):
+    """Defines a free@home switch entity."""
+
+    _attr_should_poll: bool = False
+
+    def __init__(
+        self,
+        device: SwitchActuator,
```
Comment (derjoerg) [inline custom_components/abbfreeathome_ci/select.py:101]: The other device_classes are missing

**Comment by derjoerg** [comment]
Just thinking a bit and I don't believe that this will work :smiley:
With the recent change from `isinstance` to `type` I think you need to add a specific select for each single cover (CoverActuator itself is only a super-class and doesn't exists in F@H)
```
    AtticWindowActuator
    AwningActuator
    BlindActuator
    ShutterActuator
```

**Comment by kingsleyadam** [comment]
> Just thinking a bit and I don't believe that this will work 😃 With the recent change from `isinstance` to `type` I think you need to add a specific select for each single cover (CoverActuator itself is only a super-class and doesn't exists in F@H)
> 
> ```
>     AtticWindowActuator
>     AwningActuator
>     BlindActuator
>     ShutterActuator
> ```

Yep, that's a good point, I'll adjust it to include all types of covers.

**Comment by derjoerg** [comment]
I just returned from a walk outside and wanted to share my thoughts with you :smiley:

I don't think it is good to make the `available` option dependent if the property exists in the pypi package (`available` has nothing todo with F@H). It sounds totally strange to me to create such a dependency.

I would still recommend the following:
- Make the `available` functionality dependent on a config-option like "Should forced F@H devices being marked as unavailable in HA?" (default should be FALSE)
- in the `available` property in the HA-entity, check if the config option is TRUE and if so, return the `available` status based on the forced-property of the F@H device

**Comment by kingsleyadam** [comment]
> I just returned from a walk outside and wanted to share my thoughts with you 😃
> 
> I don't think it is good to make the `available` option dependent if the property exists in the pypi package (`available` has nothing todo with F@H). It sounds totally strange to me to create such a dependency.
> 
> I would still recommend the following:
> 
> * Make the `available` functionality dependent on a config-option like "Should forced F@H devices being marked as unavailable in HA?" (default should be FALSE)
> * in the `available` property in the HA-entity, check if the config option is TRUE and if so, return the `available` status based on the forced-property of the F@H device (None and 0 == True, else False)

For now I've just decided to leave this functionality out. I think it may be confusing for users to see the entity as unavailable and not know that it's the forced position that's causing it. Further, with it unavailable you can't get the current state of the device (e.g. On/Off, Open/Closed).

I'm also unsure what setting is unavailable will do in external integrations like HomeKit or Google Assistant. Will it show as Non Responsive in HomeKit? Or Remove it completely?

The only downside to not implementing this is that the device will seem to be controllable when the forced state is enabled, but it's not.

It would be nice if Home Assistant had a Locked feature for an entity.

**Comment by kingsleyadam** [comment]
@derjoerg , I only have switches so I can't test the dimming or any of the covers. I'm not sure if you have dimming or cover actuators, but if you do could you give this a quick test?

**Comment by derjoerg** [comment]
> For now I've just decided to leave this functionality out. I think it may be confusing for users to see the entity as unavailable and not know that it's the forced position that's causing it. Further, with it unavailable you can't get the current state of the device (e.g. On/Off, Open/Closed).

Excatly

> 
> The only downside to not implementing this is that the device will seem to be controllable when the forced state is enabled, but it's not.

A note might be added in the documentation that users have to "implement" this through the dashboard.

> 
> It would be nice if Home Assistant had a Locked feature for an entity.

Absolutely, "Locked" or "Readonly"

**Comment by derjoerg** [comment]
> @derjoerg , I only have switches so I can't test the dimming or any of the covers. I'm not sure if you have dimming or cover actuators, but if you do could you give this a quick test?

I can test later today dimming and blinds

**Comment by derjoerg** [comment]
Two findings:
https://github.com/kingsleyadam/local-abbfreeathome/issues/119
And all the selects have no name
error-forced

**Comment by derjoerg** [comment]
The functionality is working as expected 👍

**Comment by kingsleyadam** [comment]
> Two findings: [kingsleyadam/local-abbfreeathome#119](https://github.com/kingsleyadam/local-abbfreeathome/issues/119) And all the selects have no name error-forced

Ok, so this probably needs the same treatment as the binary sensors.

**Comment by kingsleyadam** [comment]
> Two findings: [kingsleyadam/local-abbfreeathome#119](https://github.com/kingsleyadam/local-abbfreeathome/issues/119) And all the selects have no name error-forced

Can you try now? I appended the channel name in the translation since the device name and channel name are different, but we need the translations.

**Comment by derjoerg** [comment]
Now everything is perfect!

**Pull request merged**